core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -134,47 +153,75 @@ public void configure() throws ConfigurationException {
-			Mailbox mailbox = new Mailbox(getMailAddress());
+		if (StringUtils.isNotEmpty(emailAddress)) {
+			Mailbox mailbox = new Mailbox(emailAddress);
 			inboxId = new FolderId(WellKnownFolderName.Inbox, mailbox);
 		} else {
 			inboxId = new FolderId(WellKnownFolderName.Inbox);
 		}
 		log.debug("determined inbox ["+inboxId+"] foldername ["+inboxId.getFolderName()+"]");
 
-		if (StringUtils.isNotEmpty(getBaseFolder())) {
+		if (StringUtils.isNotEmpty(baseFolderName)) {
 			try {
-				basefolderId=findFolder(inboxId,getBaseFolder());
+				basefolderId=findFolder(inboxId,baseFolderName);
 			} catch (Exception e) {
-				throw new FileSystemException("Could not find baseFolder ["+getBaseFolder()+"] as subfolder of ["+inboxId.getFolderName()+"]", e);
+				throw new FileSystemException("Could not find baseFolder ["+baseFolderName+"] as subfolder of ["+inboxId.getFolderName()+"]", e);
 			}
 			if (basefolderId==null) {
-				log.debug("Could not get baseFolder ["+getBaseFolder()+"] as subfolder of ["+inboxId.getFolderName()+"]");
-				basefolderId=findFolder(null,getBaseFolder());
+				log.debug("Could not get baseFolder ["+baseFolderName+"] as subfolder of ["+inboxId.getFolderName()+"]");
+				basefolderId=findFolder(null,baseFolderName);
 			}
 			if (basefolderId==null) {
-				throw new FileSystemException("Could not find baseFolder ["+getBaseFolder()+"]");
+				throw new FileSystemException("Could not find baseFolder ["+baseFolderName+"]");
 			}
 		} else {
 			basefolderId=inboxId;
 		}
+		return basefolderId;
 	}
 
 
 	@Override
 	protected ExchangeService createConnection() throws FileSystemException {
 		ExchangeService exchangeService = new ExchangeService(ExchangeVersion.Exchange2010_SP2);
 
-		String defaultUsername = StringUtils.isEmpty(getAccessToken())? getUsername() : null;
-		String defaultPassword = StringUtils.isEmpty(getAccessToken())? getPassword() : getAccessToken();
+		if (client != null) {
+			ClientCredentialParameters clientCredentialParam = ClientCredentialParameters.builder(
+				Collections.singleton(SCOPE)
+			).build();
 
-		CredentialFactory cf = new CredentialFactory(getAuthAlias(), defaultUsername, defaultPassword);
-		if (StringUtils.isEmpty(cf.getUsername())) {
-			// use OAuth Bearer token authentication
-			exchangeService.getHttpHeaders().put("Authorization", "Bearer "+cf.getPassword());
+			CompletableFuture<IAuthenticationResult> future = client.acquireToken(clientCredentialParam);
+			try {
+				String token = future.get().accessToken();
+				// use OAuth Bearer token authentication
+				exchangeService.getHttpHeaders().put("Authorization", "Bearer "+token);
+			} catch (Exception e){
+				throw new FileSystemException("Could not generate access token!", e);
+			}
+			exchangeService.setImpersonatedUserId(new ImpersonatedUserId(ConnectingIdType.SmtpAddress, getMailAddress()));
+			exchangeService.getHttpHeaders().put("X-AnchorMailbox", getMailAddress());
 		} else {
+			CredentialFactory cf = getCredentials();
```

Comment:
misschien voor nu een beetje laat maar het zou wel mooier zijn als dit in configure gezet zou zijn. Deze methode wordt nu voor elke connectie aangeroepen.